src/Plutarch/SafeMoney.hs

```diff
@@ -178,10 +186,101 @@ pdiscreteValue' (Tagged (AssetClass (cs, tn))) =
             PDiscrete t <- pmatchC p
             pure $ psingletonValue # pconstant cs # pconstant tn # (pextract # t)
 
-{-
-pdiscreteValue' (Tagged (AssetClass (cs, tn))) = phoistAcyclic $
-  plam $ \p ->
-    psingletonValue
-      # pconstant cs
-      # pconstant tn
-      # Tagged.puntag p -}
+-- | @since 1.0.0
+type PRateDecimal = PFixedDecimal 1000000
+
+-- | @since 1.0.0
+newtype PExchangeRate (from :: k) (to :: k) (s :: S)
```

Comment:
This might be over-general, but I think we might want to make this even _more_ polykinded:

```haskell
newtype PExchangeRate (from :: k) (to :: k') (s :: S)
```

This might be overkill though.

Reply:
Yeah this is a good idea. I've seen some cases where one use `Symbol` for tag and other use blank datatype for tags